src/action/timeline.rs

```diff
@@ -46,22 +45,95 @@ struct RecurseState {
     encoder: GzChunkedEncoder,
 }
 
+/// Retrieves device ID from metadata.
+#[allow(unused_variables)]
+fn dev_from_metadata(metadata: &Metadata) -> u64 {
+    cfg_if! {
+        if #[cfg(target_family = "unix")] {
+            use std::os::unix::fs::MetadataExt;
+            metadata.dev()
+        } else if #[cfg(target_family = "windows")] {
+            0
+        } else {
+            compile_error!("unsupported OS family");
+        }
+    }
+}
+
+/// Tries to convert OS-dependent string to raw bytes.
+fn bytes_from_os_str(s: &OsStr) -> std::io::Result<Vec<u8>> {
+    cfg_if! {
+        if #[cfg(target_family = "unix")] {
+            use std::os::unix::ffi::OsStrExt;
+            Ok(Vec::from(s.as_bytes()))
+        } else if #[cfg(target_family = "windows")] {
+            use std::os::windows::ffi::OsStrExt;
+            use byteorder::{LittleEndian, WriteBytesExt};
+            s.encode_wide().try_fold(Vec::new(), |mut stream, ch| {
+                stream.write_u16::<LittleEndian>(ch).map(|_| stream)
+            })
```

Comment:
Interesting, I think this is a tough problem. Is the choice of using little-endian arbitrary or was there some reason for choosing it? I think it warrants a comment or perhaps a link to some resources.

Reply:
According to a quick search, UTF-16 that Windows operates with is virtually always little-endian.
Unfortunately, I can't check in practice what does Windows GRR server send to us, so I'm just assuming that it's just regular UTF16-LE without BOM.
But yeah, probably worth a comment.

Reply:
This is still missing a comment in the code to explain reasoning.